### DISCUSSION TITLE
[Yarn] First-use experience has some issues

### DISCUSSION THREAD

**Discussion by oliversalzburg** [CycloneDX/cyclonedx-node-module#310] [Q&A] [answered]
I wish I got off the ground with CycloneDX, instead I got this. Please note that the project is built with yarn.

```
$ yarn cyclonedx-bom                     
Workspace (npm lockfile v2) support is experimental. Pass `--with-workspace` flag to enable and cross your fingers. Good luck!

$ yarn cyclonedx-bom --with-workspace 
error: unknown option '--with-workspace'
```

**Comment by jkowalleck** [comment]
Described behaviour seams to be related to yarn, not this project.
Not an issue, not a question. Closed.

**Comment by oliversalzburg** [comment]
You might want to adjust the documentation of this project then, if you're "Node.js module" processor doesn't actually support NodeJS projects, but only a specific subset of them. If you _require_ the project to have been installed with a specific package manager, that should be made clear.

In case you're not aware, NodeJS ships with npm, pnpm, and yearn. Not even _anticipating_ the use of these officially supported package manager is a fault of this project.

**Comment by jkowalleck** [comment]
Does your used yarn version support workspaces? 
Are you sure you are using yarn correctly? 
Are you sure you are using cyclonedx-node correctly? how was is installed?
The mentioned error message and behavior seams to be coming from yarn, or is it caused by this project? Could you provide a reproducible setup?

**Comment by jkowalleck** [comment]
> In case you're not aware, NodeJS ships with npm, pnpm, and yearn. Not even anticipating the use of these officially supported package manager is a fault of this project.

the contrary is the case. 
have a look at the existing feature requests (issues) of the project  and see the existing integration tests.
this project supports npm and yarn-version-1 - both working as intended and has integration-tests for these environments.
this project has some issues with yarn-version-2 support, as the current list of issues show.

**Comment by oliversalzburg** [comment]
The fact that these errors are due to the lack of support for yarn above version 1 is not clear to consumers. Thanks for pointing out that you are already aware of the lack of support; the potential consumers of your project are likely not.

I was reporting on a first-use experience issue. I didn't plan to get involved in your project, read all your issues and your source to try and resolve standing issues with your implementation. If you are not open to improve your project in that regard, that is fine with me. At this point there is probably zero chance that I would be using something you built.

Have a nice day.

**Comment by jkowalleck** [comment, marked as answer]
Lets start again: 
Hello, I am the current maintainer of this project. I am not the author, builder, etc. I am just the person keeping things together.
So, your first experience was not great? Please tell what you did, what you expected to happen, what actually happened. Most importantly, please describe how you cam to the expectations, so it gets clear which documentation could be improved.
Could you elaborate what the CLI copy-paste is supposed to mean? Could you provide/describe a reproducible example/setup for it?

**Comment by jkowalleck** [comment]
this conversation got closed due to #321
